weather.py

```diff
@@ -1,104 +1,426 @@
+        min_temp = min(temp_list)
+
+        #finding what position the last occurence of min temp is at
+    occurrences = []
+    for idx, data in enumerate(temp_list):
+        if data == min_temp:
+            occurrences.append(idx)
+            last_occurence = int(occurrences[-1])          
+            
+            # have to make sure result is a TUPLE so it matches the test scripts
+    return min_temp,last_occurence
+
+# print(find_min([]))
+
+'''
+    Test cases to put into 'find_min(weather_data)' to see if the function works 
+    1. temperatures = [49, 57, 56, 55, 53] | expected_result = (49.0, 0) #PASS
+    2. temperatures = [-10, -8, 2, -16, 4] | expected_result = (-16.0, 3) #PASS
+    3. temperatures = [10.4, 14.5, 12.9, 8.9, 10.5, 11.7] | expected_result = (8.9, 3) #PASS
+    4. temperatures = ["49", "57", "56", "55", "53", "49"] | expected_result = (49.0, 5) #need code to skip the first 49 ...
+    5. temperatures = [] | expected_result = ()
+    6. temperatures = [49, 57, 56, 55, 53, 49] | expected_result = (49.0, 5)
+'''
+
+'''
+Annie's working notepad
+    temp_list = [1,2,3,4,5]
+    test_list = []
+    for temp in temp_list:
+        temp = int(temp)
+        test_list.append(temp)
+        print(test_list)
+
+    #if same value in temp_list,then skip over the first one 
+        # original_list_length = len(temp_list)
+        # no_duplicate_list = set(temp_list)
+        # print(no_duplicate_list)
+        # no_duplicate_list_length = len(no_duplicate_list)
+        # print(no_duplicate_list_length)
+        # original_list_length - duplicate_list_length > add this value to the position of min value
+'''
+
+#CHALLENGE 6 
+def find_max(weather_data):
+    """Calculates the maximum value in a list of numbers.
+
+    Args:
+        weather_data: A list of numbers.
+    Returns:
+        The maximum value and it's position in the list.
+    """
+    if weather_data == []:
+        return ()
+
+    temp_list = []
+    for temp in weather_data:
+        temp = round(float(temp),2)
+        temp_list.append(temp)
+
+    max_temp = max(temp_list)
+
+    repeats = [] 
+    for position, data in enumerate(temp_list):
+        if data == max_temp:
+            repeats.append(position)
+        
+    final_position = repeats[-1]
+
+    return max_temp, final_position
+
+# print(find_max([1.8789798,"2",3,5.5]))
+
+
+
+# CHALLENGE 7
+def generate_summary(weather_data):
+    """Outputs a summary for the given weather data.
+
+    Args:
+        weather_data: A list of lists, where each sublist represents a day of weather data.
+    Returns:
+        A string containing the summary information.
+    """
+
+## code to find title below
+    day_overview = len(weather_data) # returns how many row in list
+    # print(day_overview)
+
+    ## code to find lowest temperature
+    low_temp = []
+    for row in weather_data:
+        c = convert_f_to_c(row[1])
+        row[1] = c
+        low_temp.append(row[1])

```

Comment:
Similar to my previous comment, try to keep modified and old data stored as separate variables and avoid magic numbers. e.g.
```suggestion
    for row in weather_data:
        temp_in_celsius = convert_f_to_c(row[1])
        low_temp.append(temp_in_celsius)
```